InternetHospital.WebApi/Controllers/AppointmentsController.cs

```diff
@@ -44,92 +42,56 @@ public IActionResult GetAvailableAppointments([FromQuery] int doctorId)
         [HttpPost("cancel")]
         public IActionResult CancelAppointment([FromBody] AppointmentEditingModel model)
         {
-            if (int.TryParse(User.Identity.Name, out var doctorId))
-            {
-                var (status, message) = _appointmentService.CancelAppointment(model.Id, doctorId);
-
-                if (status)
-                {
-                    return Ok(new { message });
-                }
-                else
-                {
-                    return BadRequest(new { message });
-                }
-            }
-            else
+            if (!int.TryParse(User.Identity.Name, out var doctorId))
             {
                 return BadRequest(new { message = "Wrong claims" });
             }
+
+            var (status, message) = _appointmentService.CancelAppointment(model.Id, doctorId);
+
+            return status ? (IActionResult)Ok(new { message }) : BadRequest(new { message });
         }
 
         [Authorize(Policy = "ApprovedDoctors")]
-        [HttpPost("delete")]
+        [HttpDelete("delete")]
         public IActionResult DeleteAppointment([FromBody] AppointmentEditingModel model)
         {
-            if (int.TryParse(User.Identity.Name, out var doctorId))
-            {
-                var (status, message) = _appointmentService.DeleteAppointment(model.Id, doctorId);
-
-                if (status)
-                {
-                    return Ok(new { message });
-                }
-                else
-                {
-                    return BadRequest(new { message });
-                }
-            }
-            else
+            if (!int.TryParse(User.Identity.Name, out var doctorId))
             {
                 return BadRequest(new { message = "Wrong claims" });
             }
+
+            var (status, message) = _appointmentService.DeleteAppointment(model.Id, doctorId);
+
+            return status ? (IActionResult)Ok(new { message }) : BadRequest(new { message });
         }
 
         [Authorize(Policy = "ApprovedDoctors")]
         [HttpPost("finish")]
         public IActionResult FinishAppointment([FromBody] AppointmentEditingModel model)
         {
-            if (int.TryParse(User.Identity.Name, out var doctorId))
-            {
-                var (status, message) = _appointmentService.FinishAppointment(model.Id, doctorId);
-
-                if (status)
-                {
-                    return Ok(new { message });
-                }
-                else
-                {
-                    return BadRequest(new { message });
-                }
-            }
-            else
+            if (!int.TryParse(User.Identity.Name, out var doctorId))
             {
                 return BadRequest(new { message = "Wrong claims" });
             }
+
+            var (status, message) = _appointmentService.FinishAppointment(model.Id, doctorId);
+
+            return status ? (IActionResult)Ok(new { message }) : BadRequest(new { message });
```

Comment:
Here the same question.

Reply:
Here the same answer: There is no implicit conversion between OkObjectResult and BadRequestObjectResult